### PR TITLE
Keep credit order

### PR DESCRIPTION
**Pull request by tricknotes**
ref: https://github.com/bm-sms/nursepress/issues/161

### CODE REVIEW

**Comment by tricknotes** [comment]
When this PR become ready to merge, I'll make another PR with only migration.

File: app/models/credit.rb [hunk 1]
```diff
@@ -5,4 +5,12 @@ class Credit < ActiveRecord::Base
 
   validates :participant_id, presence: true
   validates :credit_role_id, presence: true
+
+  before_save :setup_order
+
+  private
+
+  def setup_order
+    self.order ||= (post.credits.map(&:order).compact.max || 0) + 1
```
Comment (mtsmfm) [inline app/models/credit.rb]: `post.credits.maximum(:order) || 0`

Comment (tricknotes) [inline app/models/credit.rb]: I want to consider about not persisted records.

File: app/views/editor/posts/_form.html.haml [hunk 2]
```diff
@@ -30,10 +30,24 @@
...
+      var maxOrder = Math.max.apply(null, orders);
+      $element.find('.credit-order-value').val(maxOrder + 1);
+    }
+
     applySelect2($('select#post_category_id'), {allowClear: false});
     applySelect2($('select#post_serial_id'), {allowClear: true});
 
     $(document).on('cocoon:after-insert', function(e, inserted) {
-      applySelect2($(inserted[0]).find('select'), {allowClear: false});
+      let $element = $(inserted[0]);
```
Comment (mtsmfm) [inline app/views/editor/posts/_form.html.haml]: Can we use `let`?

Comment (tricknotes) [inline app/views/editor/posts/_form.html.haml]: This code is not built by Babel due to view file...
Should we move this to under `app/assets`?

File: app/models/credit.rb (hunk 1 above)
Comment (mtsmfm) [inline app/models/credit.rb]: Hmm, when should we consider about that?
Could you tell me more detail?

Comment (tricknotes) [inline app/models/credit.rb]: Such as rails console or test.
However this code is unnecessary for production code.

I'll drop `Credit#setup_order`.

**Comment by mtsmfm** [comment]
:shipit:

**Pull request merged**